### PR TITLE
Add openapi docs for OpenAI api version 2023-08-01-preview

### PR DESCRIPTION
**Pull request by wangyuantao**
# Data Plane API - Pull Request
New api version 2023-08-01-preview for open ai. see diff via `diff 2023-07-01-preview/inference.json 2023-08-01-preview/inference.json  > diff.txt`

```
6c6
<     "version": "2023-07-01-preview"
---
>     "version": "2023-08-01-preview"
50c50
<               "example": "2023-07-01-preview",
---
>               "example": "2023-08-01-preview",
168,172d167
<                   "cache_level": {
<                     "description": "can be used to disable any server-side caching, 0=no cache, 1=prompt prefix enabled, 2=full cache",
<                     "type": "integer",
<                     "nullable": true
<                   },
370c365
<               "example": "2023-07-01-preview",
---
>               "example": "2023-08-01-preview",
511c506
<               "example": "2023-07-01-preview",
---
>               "example": "2023-08-01-preview",
586c581
<               "example": "2023-07-01-preview",
---
>               "example": "2023-08-01-preview",
1195a1191,1206
>           },
>           "context": {
>             "type": "object",
>             "description": "The conversation context",
>             "nullable": true,
>             "properties": {
>               "messages": {
>                 "type": "array",
>                 "description": "Messages exchanged between model and extensions prior to final message from model",
>                 "minItems": 1,
>                 "items": {
>                   "$ref": "#/components/schemas/message"
>                 },
>                 "nullable": true
>               }
>             }
1265,1270c1276,1278
<               "messages": {
<                 "type": "array",
<                 "description": "The list of messages returned by the service.",
<                 "items": {
<                   "$ref": "#/components/schemas/message"
<                 }
---
>               "message": {
>                 "description": "The message returned by the service.",
>                 "$ref": "#/components/schemas/message"
1303,1312c1311,1322
<               "messages": [
<                 {
<                   "role": "tool",
<                   "content": "{\"citations\":[{\"filepath\":\"ContosoTraveler.pdf\",\"content\":\"This is the content of the citation 1\"},{\"filepath\":\"WestCoastTraveler.html\",\"content\":\"This is the content of the citation 2\"},{\"content\":\"This is the content of the citation 3 without filepath\"}],\"intent\":\"hiking place in seattle\"}",
<                   "end_turn": false
<                 },
<                 {
<                   "role": "assistant",
<                   "content": "Seattle is a great place for hiking! Here are some of the best hiking places in Seattle according to Contoso Traveler [doc1] and West Coast Traveler, Snow Lake, Mount Si, and Mount Tenerife [doc2]. I hope this helps! Let me know if you need more information.",
<                   "end_turn": true
---
>               "message": {
>                 "role": "assistant",
>                 "content": "Seattle is a great place for hiking! Here are some of the best hiking places in Seattle according to Contoso Traveler [doc1] and West Coast Traveler, Snow Lake, Mount Si, and Mount Tenerife [doc2]. I hope this helps! Let me know if you need more information.",
>                 "end_turn": true,
>                 "context": {
>                   "messages": [
>                     {
>                       "role": "tool",
>                       "content": "{\"citations\":[{\"filepath\":\"ContosoTraveler.pdf\",\"content\":\"This is the content of the citation 1\"},{\"filepath\":\"WestCoastTraveler.html\",\"content\":\"This is the content of the citation 2\"},{\"content\":\"This is the content of the citation 3 without filepath\"}],\"intent\":\"hiking place in seattle\"}",
>                       "end_turn": false
>                     }
>                   ]
1314c1324
<               ]
---
>               }

```

## API Info: The Basics
https://github.com/Azure/azure-rest-api-specs-pr/pull/13558

* Link to API Spec engagement record issue:
https://github.com/Azure/azure-rest-api-specs/issues/24733

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/inference.json [hunk 1]
```diff
@@ -0,0 +1,1373 @@
...
+          }
+        ],
+        "requestBody": {
+          "required": true,
+          "content": {
+            "application/json": {
+              "schema": {
+                "type": "object",
+                "properties": {
+                  "messages": {
```
Comment (johanste) [inline specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/inference.json:518]: are functions gone? They were included in the 2023-07-01-preview API versions....

Comment (trrwilson) [reply specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/inference.json:518]: definitely still supported and agreed that functions changes need to be restored.
RAI content filter results appear to be in there so it's likely just 2023-07-01-preview deltas (which should just be functions) that weren't ported -- though re-checking a full diff is still a good idea.

Comment (wangyuantao) [reply specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/inference.json:518]: Thanks for catching it! The functions were not there when I reviewed the PR with the board, that is why they are missing. Let me cherry pick the functions related changes to Aug.

**Review by johanste — APPROVED** [review]

**Comment by johanste** [comment]
The breaking changes are only for response properties (specifically for error models). Making them read-only is not breaking here.

**Pull request merged**